pkg/controller/replicaset/replica_set_utils.go

```diff
@@ -22,15 +22,25 @@ import (
 	"context"
 	"fmt"
 	"reflect"
+	"time"
 
 	"k8s.io/klog/v2"
 
 	apps "k8s.io/api/apps/v1"
 	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	appsclient "k8s.io/client-go/kubernetes/typed/apps/v1"
 	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
+	"k8s.io/kubernetes/pkg/features"
+)
+
+const (
+	// AllReplicasAvailableReason is added in a replica set when all of its replicas are available.
+	AllReplicasAvailableReason = "AllReplicasAvailable"
+	// AllReplicasUnavailableReason is added in a replica set when some of its replicas are unavailable.
+	AllReplicasUnavailableReason = "AllReplicasUnavailable"
```

Comment:
The naming of reasons is a change compared to the Deployment (which has `MinimumReplicasAvailable`/`MinimumReplicasUnavailable` reasons). Since we only acknowledge full replica set availability or not I named it `AllReplicasAvailable`/`AllReplicasUnavailable`.  Although the second one would probably read better as `SomeReplicaUnavailable`.

Thoughts?